x-pack/plugins/fleet/server/services/agents/versions.ts

```diff
@@ -60,8 +60,7 @@ export const getAvailableVersions = async ({
       .sort((a: any, b: any) => (semverGt(a, b) ? -1 : 1));
     versionsToDisplay = uniq(versions) as string[];
 
-    const appendCurrentVersion =
-      includeCurrentVersion ?? !config?.internal?.onlyAllowAgentUpgradeToKnownVersions;
+    const appendCurrentVersion = includeCurrentVersion;
```

Comment:
changed this not to include kibana version in stateful neither, as it's not correct for unreleased versions:
<img width="1306" alt="image" src="https://github.com/elastic/kibana/assets/90178898/cdeede00-dc69-47a2-adbf-2aee2ede7494">